src/controllers/controller.bar.js

```diff
@@ -342,10 +342,14 @@ module.exports = function(Chart) {
 			rightCategorySize = rightSampleSize * options.categoryPercentage;
 			fullBarSize = (leftCategorySize + rightCategorySize) / ruler.stackCount;
 			size = fullBarSize * options.barPercentage;
+			size = helpers.valueOrDefault(options.barThickness, size);
 
-			size = Math.min(
-				helpers.valueOrDefault(options.barThickness, size),
-				helpers.valueOrDefault(options.maxBarThickness, Infinity));
+			if (size > helpers.valueOrDefault(options.maxBarThickness, Infinity)) {
```

Comment:
I'm not sure to get the logic here, can you describe what this code is trying to solve?

Reply:
`leftCategorySize` is the part of the bar drawn to the left of the tick and `rightCategorySize` to the right. The old code made `size` smaller without making either `leftCategorySize` or `rightCategorySize` smaller which doesn't work. If you make `size` smaller then you need to make the left and right hand sides smaller accordingly

Reply:
Makes sense, but I think that `leftCategorySize` and `rightCategorySize` should be equal if  `options.barThickness` is specified, and thus centered around the tick mark.

Reply:
Imagine you originally have left = 3, right = 17 and max = 10. If you try to make them the same then you end up making the left segment larger than it was originally and it can overlap with another bar.

I think if we want to center the bar on the tick then we should always do it and not just in this one case.

Reply:
I'm not sure I agree on that last statement because when `barThickness` is specified, I think it makes sense to have it centered on the tick mark, even if it overlaps. When no explicit size specified, we compute a bar size that naturally fit between previous and next ticks (so not centered in case of irregular data).

Reply:
That would be introducing a new behavior. I don't think the behavior makes sense now, but if we're going to change it I'd rather do that in 2.8.0 than in a minor point release

Reply:
In this case, I think it make sense to implement the correct behaviour and retarget this to 2.8.0. Avoids having to come back here again almost immediately and if someone depends on the 2.7.1 behaviour we've introduced an issue for them when 2.8 releases.

If we really want to keep both behaviours, a new option could be added to control the bar alignment (centered vs auto-fit).

Reply:
Ok. I will move the issue to 2.8.0

I'll close this PR for now given that the eventual solution will be different